### PR TITLE
Trait framework

### PR DESCRIPTION
**Pull request by bernie-g**
This PR was roughly based off of Mekanism's [Attributes](https://github.com/mekanism/Mekanism/blob/793be440035c4cd965ac2a5120ed1436fd663f4d/src/main/java/mekanism/common/block/attribute/Attributes.java)

This PR creates an abstracted Trait system we can use for blocks, items, tile entities, etc. to create new blocks by composing traits together and avoiding the [diamond inheritance problem](https://en.wikipedia.org/wiki/Multiple_inheritance#The_diamond_problem). This also should help us solve some interesting problems like block description tooltips, slave blocks, and multiblocks.

The main feature of this PR are `Behaviour`'s and `Trait`'s. A trait can do essentially anything, but the useful part is `tweakers`. Trait's can register a `Tweaker` that operates on a specific type of object. For example, you could register a trait that tweaks BlockProperties, and then whenever a block is setting up its properties it will go through all the traits and apply its tweaks to the properties. This is useful for composing features.

A behaviour is a collection of traits with some extra stuff. Behaviours can be created using the appropriate Builder class and are composable. The builder allows you to add traits, remove traits, and compose behaviours together to copy over all the traits. Behaviours also let you set required traits, using `Behaviour.Builder.requiredTrait()`. This ensures that a trait of a specific type has been added to a Behaviour, and throws an exception if it has not. This is useful for requiring certain traits for behaviours; for example you could set the BlockProperties trait to required for a `MachineBehaviour`, but then let it be implemented in the composed `BotariumBehaviour`. If you forget to implement it, it throws an exception.

This PR replaces some block mechanics with the trait system but it'd be nice to replace more as we go along, such as voxel shapes and interaction with blocks. Eventually tile entities and guis should also use traits.

### CODE REVIEW

File: src/main/java/software/bernie/techarium/trait/block/BlockBehaviours.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package software.bernie.techarium.trait.block;
+
+import software.bernie.techarium.registry.LangRegistry;
+import software.bernie.techarium.tile.arboretum.ArboretumTile;
+import software.bernie.techarium.tile.botarium.BotariumTile;
+import software.bernie.techarium.tile.exchangestation.ExchangeStationTile;
+
+public class BlockBehaviours {
```
Comment (agnor99) [inline src/main/java/software/bernie/techarium/trait/block/BlockBehaviours.java:8]: Make them final and CAPS LOCK all of them

Comment (agnor99) [reply src/main/java/software/bernie/techarium/trait/block/BlockBehaviours.java:8]: Also add an automatable behaviour/Trait for the Assembler, so it doesn't expose item capability, if the trait/behaviour is missing

**Review by agnor99 — APPROVED** [review]

File: src/main/java/software/bernie/techarium/trait/block/BlockBehaviours.java (hunk 1 above)
Comment (bernie-g) [reply src/main/java/software/bernie/techarium/trait/block/BlockBehaviours.java:8]: I'm not sure what you mean by that

Comment (agnor99) [reply src/main/java/software/bernie/techarium/trait/block/BlockBehaviours.java:8]: Nvmd, me dumb at 7am

**Review by 50ap5ud5 — APPROVED** [review]
Looks good, changes are well documented and cleanly executed.

**Pull request merged**